includes/Shopping/Shopify_Query.php

```diff
@@ -231,9 +231,32 @@ protected function fetch_remote_products( string $search_term ) {
 		 */
 		$result = json_decode( $body, true );
 
-		// TODO(#11268): Error handling.
 		if ( isset( $result['errors'] ) ) {
-			return new WP_Error( 'rest_unknown', __( 'Error fetching products', 'web-stories' ), [ 'status' => 404 ] );
+			try {
+				$error      = array_shift( $result['errors'] );
+				$error_code = $error['extensions']['code'];
```

Comment:
I would check if this exists / isset and default to an empty string. 
```suggestion
				$error_code = $error['extensions']['code'] ?? '';
```